docs/topics/auth/username.txt

```diff
@@ -0,0 +1,140 @@
+----------------------------
+
+Username validation is configured in the
+:setting:`AUTH_USERNAME_VALIDATORS` setting::
+
+    AUTH_USERNAME_VALIDATORS = [
+        {
+            'NAME': 'django.contrib.auth.validators.UnicodeUsernameValidators',
+        },
+        {
+            'NAME': 'django.contrib.auth.validators.UsernameMinimumLengthValidator',
+            'OPTIONS':{
+                'min_length':3,
+            }
+        }
+    ]
+
+This example enables two included validators:
+
+* ``UnicodeUsernameValidator``, which checks whether username contains
+  anything other than the English letters, numbers and @/./+/-/_.
+* ``UsernameMinimumLengthValidator``, which checks whether the username meets a minimum
+  length. This validator is configured with a custom option it now requires
+  the minimum length to be three characters instead of the default one.
+
+The help texts and any errors from username validators are always returned in
+the order they are listed in :setting:`AUTH_USERNAME_VALIDATORS`.
+
+Included validators
+-------------------
+
+Django includes two validators:
+
+.. class:: UnicodeUsernameValidators
+    
+    Validates whether the username contains Unicode characters, in addition
+    to ``@``,``.``,``+``,``-``,and ``_``. This is the default validator.
+
+.. class:: ASCIIUsernameValidators
+
+    Validates whether the username contains ASCII letters, in addition
+    to ``@``,``.``,``+``,``-``,and ``_``. 
+
+.. class:: UsernameMinimumLengthValidator(min_length=1)
+
+.. versionadded:: 3.2
+
+    Validates whether the username meets a minimum length.
+    The minimum length can be customized with the ``min_length`` parameter.
+
+Integrating validation
+----------------------
+
+There are few functions in ``django.contrib.auth.username_validation`` that
+you can call from your own forms or other code to integrate username
+validation.
+
+.. function:: username_validators_help_texts(username_validators=None)
+
+.. versionadded:: 3.2
+
+    Returns a list of the help texts of all validators. These explain the
+    username requirements to the user.
+
+.. function:: username_validators_help_text_html(username_validators=None)
+
+.. versionadded:: 3.2
+
+    Returns an HTML string with all the text in an ``<ul>``. This is
+    helpful when adding username validation to forms, as you can pass the
+    output directly to the ``help_text`` parameter of a form field.
+
+Writing your own validator
+--------------------------
+
+If Django's built-in validators are not sufficient for your requirements,
+you can write your own username validators. Validators have a fairly small
+interface. They must must implement two methods.
+
+* ``__call__(self, username)``: validate a username. Return ``None`` if the
+  username is valid, or raise a 
+  :exc:`~django.core.exceptions.ValidationError` with an error message if the
+  username is not valid.
+* ``get_help_text(self)``: provide a help text to explain the requirements to
```

Comment:
Are you sure `get_help_text(self)` works? The validators you contribute have a method called `help_text(self)` instead. The documentation should be consistent, but more importantly, I suggest writing a test that implements a custom validator so that we would have caught this already.